test/Composer/ComposerJsonSpec.hs

```diff
@@ -0,0 +1,75 @@
+import Types (License (..), LicenseResult (..), LicenseType (LicenseSPDX))
+
+spec :: Spec
+spec = do
+  currentDir <- runIO getCurrentDir
+  describe "composer.json LicenseAnalyzeProject" $ do
+    it' "Reads a composer.json with a single license string" $
+      let baseDir = currentDir </> $(mkRelDir "test/Composer/testdata/composer_json_single")
+          project =
+            ComposerProject
+              { composerDir = baseDir
+              , composerLock = currentDir </> $(mkRelFile "not_tested_here")
+              }
+       in do
+            licenses <- licenseAnalyzeProject project
+            licenses
+              `shouldBe'` [ LicenseResult
+                              { licenseFile = toFilePath $ baseDir </> $(mkRelFile "composer.json")
+                              , licensesFound =
+                                  [ License
+                                      { licenseType = LicenseSPDX
+                                      , licenseValue = "MIT"
+                                      }
+                                  ]
+                              }
+                          ]
+
+    it' "Reads a composer.json with multiple license strings" $
+      let baseDir = currentDir </> $(mkRelDir "test/Composer/testdata/composer_json_many")
+          project =
+            ComposerProject
+              { composerDir = baseDir
+              , composerLock = currentDir </> $(mkRelFile "not_tested_here")
+              }
+       in do
+            licenses <- licenseAnalyzeProject project
+            licenses
+              `shouldBe'` [ LicenseResult
+                              { licenseFile = toFilePath $ baseDir </> $(mkRelFile "composer.json")
+                              , licensesFound =
+                                  [ License
+                                      { licenseType = LicenseSPDX
+                                      , licenseValue = "MIT"
+                                      }
+                                  , License
+                                      { licenseType = LicenseSPDX
+                                      , licenseValue = "GPL"
+                                      }
+                                  , License
+                                      { licenseType = LicenseSPDX
+                                      , licenseValue = "BSD"
+                                      }
+                                  ]
+                              }
+                          ]
+
+    it "Fails to read a composer.json which is out of schema" $
+      let baseDir = currentDir </> $(mkRelDir "test/Composer/testdata/composer_json_bad_schema")
+          project =
+            ComposerProject
+              { composerDir = baseDir
+              , composerLock = currentDir </> $(mkRelFile "not_tested_here")
+              }
+       in do
+            shouldThrow (runTestEffects $ licenseAnalyzeProject project >> pure ()) anyException
```

Comment:
Ideally, we should use our normal error handling to test this.  I believe that `readContentsJson` returns `Diagnostics.fatal` if there is a parsing error.  We can use `Diagnostics.recover` to catch that, which performs `m a -> m (Maybe a)`, returning `Nothing` if there was some error.  Checking for `Nothing` here would catch this properly.